### ISSUE TITLE
`entity` URL omits hostname for non-top 25 TLDs

### ISSUE THREAD

**Issue by benschwarz** [GoogleChrome/lighthouse#15623]
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://www.hydro.mb.ca

### What happened?

Lighthouse [uses a function](https://github.com/GoogleChrome/lighthouse/blob/main/shared/util.js#L314) to "cleanly" format URLs (e.g.: www.mydomain.tld → mydomain.tld), but this is flawed because it uses a [hard coded list of top 25 TLDs.](https://github.com/GoogleChrome/lighthouse/blob/main/shared/util.js#L25C7-L25C17).

In the case of any TLD outside of that list, Lighthouse will incorrectly shorten URLs, which presents in `entity` attributes within JSON output: 

<img width="1127" alt="image" src="https://github.com/GoogleChrome/lighthouse/assets/924/1843b649-756e-418b-a7c4-a860866469cf">

### What did you expect?

Expected the hostname to be included within `entity`.

### How were you running Lighthouse?

CLI, node

### Lighthouse Version

11+

### Chrome Version

N/A

### Node Version

N/A

### OS

N/A

**Comment by adamraine** [comment]
To fix this completely we would need to reqork the tld function to account for every entry in https://publicsuffix.org/list/public_suffix_list.dat. That would add a considerable amount of data to the Lighthouse bundle. Considering this function was added in 2018 and it hasn't been an issue until now we don't think this is worth doing.

We can expand the list of tlds (maybe top 50?) as a half-measure for now though.

**Comment by benschwarz** [comment]
> To fix this completely we would need to reqork the tld function to account for every entry in https://publicsuffix.org/list/public_suffix_list.dat.

That's true, but only if it's fixed by using the same method of having a hard coded list of TLDs. Catering for top 25 (predominantly wealthy western) TLDs is a flawed approach, perhaps there's an alternative solution for formatting a URL cleanly? I'd be happy to PR if there's another tenable idea. 

> That would add a considerable amount of data to the Lighthouse bundle.

At the time of writing, the public suffix DAT file is 298kb uncompressed, 81kb gzipped.

> Considering this function was added in 2018 and it hasn't been an issue until now we don't think this is worth doing.

It's the first time it's been reported, that's true, but as LH continues to add `entity` attributes throughout the JSON report, it's more likely to be an observed issue.

**Comment by alexnj** [comment]
I tried to find a reasonably optimized PSL implementation that's also maintained, and prototyped the changes required to bring it in. Looks like we'll add 40 KiB compressed (and 100 KiB uncompressed). 

I realized the primary consumer of TLD on the reporting side is 3p-classification, so I rewrote it to rely on entity-classification lookup (could be optimized with the LHR TLD table @connorjclark mentioned in our discussion, if needed). We'll still need to retain the old 25-TLD based approach to render pre-10.0 reports that have no entity classification. 

Smaller footprint might be achievable if we can reuse the Chrome PSL while running within the browser, but I'm not sure how involved that is yet.
